File: selenium_UI/src/main/java/com/cisco/appko/automation/selenium/pages/facebook/FbLoginPage.java
package com.cisco.appko.automation.selenium.pages.facebook;

import com.cisco.appko.automation.selenium.pages.core.Page;
import com.cisco.appko.automation.user.BaseUser;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.testng.Assert;
import ru.yandex.qatools.allure.annotations.Step;

import static com.cisco.appko.automation.webdriver.ElementsUtil.isElementPresent;
import static com.cisco.appko.automation.webdriver.ElementsUtil.waitForPageLoaded;

/**
 * Created by eugene polschikov on 2018-03-11.
 */
public class FbLoginPage extends Page {
    public FbLoginPage(WebDriver driver) {
        super(driver);
    }

    private final static String EMAIL_INPUT_CSS = "#email";
    private final static String PASSWORD_INPUT_CSS = "#pass";
    private final static String LOGIN_BUTTON_CSS = "#loginbutton";

    @FindBy(css = EMAIL_INPUT_CSS)
    private WebElement emailInput;

    @FindBy(css = PASSWORD_INPUT_CSS)
    private WebElement passwordInput;

    @FindBy(css = LOGIN_BUTTON_CSS)
    private WebElement loginButton;

    @Step("login into facebook")
    public FbAuthorizedUserPage facebookLogin(BaseUser user) {

        log.info("checking that login input is present on the page. If Login input is not " +
                "present on the page -> test execution is stopped.");
        Assert.assertTrue(emailInput.isDisplayed(), "ooops, login input is NOT on the page");

        log.info("entering login");
        emailInput.clear();
        emailInput.sendKeys(user.getEmail());


        log.info("checking that password input is present on the page");
        Assert.assertTrue(passwordInput.isDisplayed(), "ooops, login input is NOT on the page");
        log.info("entering password");
        passwordInput.clear();
        passwordInput.sendKeys(user.getPassword());


        log.info("checking that login button is on the page");
        Assert.assertTrue(isElementPresent(driver, By.cssSelector(LOGIN_BUTTON_CSS)), "ooops. login button" +
                "is absent on the page");
        log.info("pressing submit/sign-in button");
        loginButton.click();

        waitForPageLoaded(driver);
        return new FbAuthorizedUserPage(driver);
    }


}

File: webdriverPhp/pages/FbLoginPage.php
<?php

namespace Pages;

use core\BaseUser;
use Core\WebdriverInit;
use Core\WebdriverUtils;
use Facebook\WebDriver\Exception\NoSuchElementException;
use Facebook\WebDriver\Exception\TimeOutException;
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\WebDriverBy;
use Facebook\WebDriver\WebDriverExpectedCondition;

class FbLoginPage
{
    private $emailInputCss = "#email";
    private $passwordInputCss = "#pass";
    private $loginButtonCss = "#loginbutton";


    /**
     * @return string
     */
    public function getEmailInputCss(): string
    {
        return $this->emailInputCss;
    }


    /**
     * @return RemoteWebDriver|null
     */
    public function getDriver(): ?RemoteWebDriver
    {
        return $this->driver;
    }


    var $driver = null;

    /**
     * FacebookLoginPage constructor.
     * @param $driver
     */
    public function __construct(RemoteWebDriver $driver)
    {
        $this->driver = $driver;
    }

    public function facebookLanding()
    {

        // initial navigation to facebook
        $this->driver->get('http://www.facebook.com');
//        print_r($this->driver->manage()->getCookies());
        print_r("###wait until page title contains 'Facebook'###");
        // Wait for at most 10s and retry every 500ms if it the title is not correct.
        try {
            $this->driver->wait(WebdriverInit::getDefaultTimeout(), 500)->until(
                WebDriverExpectedCondition::titleContains("Facebook")
            );
        } catch (NoSuchElementException $e) {
            print_r("ERROR! No such element have been found!");
        } catch (TimeOutException $e) {
            print_r("ERROR! Timeout reached!");
        } catch (\Exception $e) {
            print_r("ERROR! Please contact test developers for investigation!");
        }


        return $this;
    }

    /**
     * @param BaseUser $user
     * @return mixed
     */
    public function facebookLogin(BaseUser $user)
    {

        $emailInput = $this->driver->findElement(WebDriverBy::cssSelector($this->emailInputCss));
        WebdriverUtils::fillInput($emailInput, $user->getEmail());
//password input init
        $passwordInput = $this->driver->findElement(WebDriverBy::cssSelector($this->passwordInputCss));

        WebdriverUtils::fillInput($passwordInput, $user->getPassword());

//submit button init0
        $submitButton = $this->driver->findElement(WebDriverBy::cssSelector($this->loginButtonCss));
        $submitButton->click();

        $this->driver->wait(2500);


        echo "response on element presence goes below";
        print_r(var_export(WebdriverUtils::isElementPresent($this->driver, $this->loginButtonCss)));

        return new FbAuthorizedLandingpage($this->driver);

    }

}


//exit;

//restore:

//var_export($driver);
//var_export($driver->manage()->getCookies());
//file_put_contents($file, serialize($driver));



